src/DotLiquid.Tests/ContextTests.cs

```diff
@@ -785,34 +790,78 @@ public void TestNestedDynamicAsVariable()
             Assert.AreEqual("Hello", _context["dynamic.lambda.name"]);
         }
 
+        private class ExpandoModel
+        {
+            public int IntProp { get; set; }
+            public string StrProp { get; set; }
+            public ExpandoObject Props { get; set; }
+        }
+
+        // Test case for [Issue #350](https://github.com/dotliquid/dotliquid/issues/350)
+        [Test]
+        public void TestNestedExpandoTemplate_Issue350()
+        {
+            var model = new ExpandoModel()
+            {
+                IntProp = 23,
+                StrProp = "from string prop",
+                Props = new ExpandoObject()
+            };
+            var dic = (IDictionary<string, object>)model.Props;
+            dic.Add("Key1", "from expando prop");
+
+            Template.RegisterSafeType(typeof(ExpandoModel), new[] { "IntProp", "StrProp", "Props" });
+            const string templateString = @"Int: '{{IntProp}}'; String: '{{StrProp}}'; Expando: '{{Props.Key1}}'";
+            var t = Template.Parse(templateString);
+            Assert.AreEqual(expected: "Int: '23'; String: 'from string prop'; Expando: 'from expando prop'",
+                            actual: t.Render(Hash.FromAnonymousObject(model)));
+        }
+
+        private class DataJson : Drop
+        {
+            public Dictionary<string, object> Data { get; set; }
+        }
+
+        // Test case for [Issue #417](https://github.com/dotliquid/dotliquid/issues/417)

```

Comment:
Nit: non-standard comment format at this position. `//` should be used for in-method comments above a line of code. Above a method the XML comment format `/// <summary>...` should be used.

Reply:
Fixed.